resdk/analysis/prepare_geo.py

```diff
@@ -0,0 +1,166 @@
+    """Run ``Prepare GEO - ChIP-Seq`` process on the resource.
+
+    This method can be used to run ``Prepare GEO - ChIP-Seq`` process
+    on a single collection or a list of samples.
+
+    param str name: name of the prepare GEO tarball and table
+
+    """
+
+    reads = []
+    macs14 = []
+    relations = []
+
+    samples = get_samples(resource)
+    resolwe = get_resolwe(*samples)
+    collection_ids = set()
+
+    for sample in samples:
+        reads.append(sample.get_reads().id)
+        if is_background(sample):
+            continue
+
+        macs_list = sample.get_macs()
+        if not macs_list:
+            raise TypeError("Sample {} has no `macs14` data "
+                            "object!".format(sample))
+
+        if len(macs_list) == 1:
+            macs14.append(macs_list[0].id)
+        else:
+            raise TypeError("Sample {} has more than one `macs14` "
+                            "data objects!".format(sample))
+
+        background = sample.get_background(fail_silently=True)
+        if background:
+            if background in samples:
+                relations.append(
+                    ':'.join([sample.name, background.name])
+                )
+            else:
+                raise TypeError("Background of the sample {0} cannot "
+                                "be found in the resource you provided"
+                                ": {1}!".format(sample, resource))
+
+        collection_ids.add(get_resource_collection(sample))
+
+    new_name, collection = get_name_collection(collection_ids, resolwe)
+    if not name:
+        name = new_name
+
+    inputs = {
+        'reads': reads,
+        'macs14': macs14,
+        'relations': relations,
+        'name': name,
+    }
+    geo = resolwe.get_or_run(slug='prepare-geo-chipseq', input=inputs)
+
+    if collection:
+        collection.add_data(geo)
+
+    return geo
+
+
+def prepare_geo_rnaseq(resource, name=None):
+    """Run ``Prepare GEO - RNA-Seq`` process on the resource.
+
+    This method can be used to run ``Prepare GEO - RNA-Seq`` process
+    on a single collection or a list of samples.
+
+    param str name: name of the prepare GEO tarball and table
+
+    """
+
+    reads = []
+    expressions = []
+
+    samples = get_samples(resource)
+    resolwe = get_resolwe(*samples)
+    collection_ids = set()
+
+    for sample in samples:
+        reads.append(sample.get_reads().id)
+        expressions.append(sample.get_expression().id)
+        collection_ids.add(get_resource_collection(sample))
+
+    new_name, collection = get_name_collection(collection_ids, resolwe)
+    if not name:
+        name = new_name
+
+    inputs = {
+        'reads': reads,
+        'expressions': expressions,
+        'name': name,
+    }
+    geo = resolwe.get_or_run(slug='prepare-geo-rnaseq', input=inputs)
+
+    if collection:
+        collection.add_data(geo)
+
+    return geo
+
+
+def prepare_geo(resource, types=[], name=None):
```

Comment:
You can define this as `def prepare_geo(resource, name=None, *types):`, so you can call `prepare_geo('rna_seq')` instead of `prepare_geo(['rna_seq'])`.